package/MDAnalysis/core/selection.py

```diff
@@ -261,6 +264,15 @@ def apply(self, group):
         # All atoms in group that aren't in sel
         sys = group[~np.in1d(group.indices, sel.indices)]
 
+        if self.cutoff == 0.0:
```

Comment:
It's pedantic, but this misses the case of atoms in different images at the same position.  I'd sooner actually fix what is causing the segfault than try and special-case it.

Reply:
Can you provide an example unit test for that case? If these things are genuinely important there probably should be unit tests already in place for them.

Reply:
Would something like this work?
```python
def test_around_superposed_atoms(self):
    u = mda.Universe(PDB)
    u.dimensions = [1e-2, 1e-2, 1e-2, 90, 90, 90]
    ag = u.select_atoms('around 0.0 resid 1')
    assert len(ag) == len(u.atoms)-len(u.residues[0].atoms)
```

from below:
```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PDB
>>> u = mda.Universe(PDB)
>>> u.select_atoms('around 0.1 resid 1')
<AtomGroup with 0 atoms>
>>> u.dimensions = [1e-2, 1e-2, 1e-2, 90, 90, 90]
>>> u.select_atoms('around 0.1 resid 1')
<AtomGroup with 47662 atoms>
>>> u.atoms.wrap()
array([[6.9999695e-03, 3.8146973e-06, 5.0010681e-03],
       [7.9994202e-03, 1.9989014e-03, 2.0008087e-03],
       [9.9945068e-04, 7.9994202e-03, 9.0007782e-03],
       ...,
       [2.0065308e-03, 1.9989014e-03, 7.9994202e-03],
       [3.9978027e-03, 4.0016174e-03, 4.0006638e-03],
       [9.9945068e-04, 8.9988708e-03, 3.0000210e-03]], dtype=float32)
>>> pos = [tuple(x) for x in u.atoms.positions]
>>> pos[1] in pos[len(u.residues[0].atoms):]
True
>>> u.select_atoms('around 0.1 resid 1')
<AtomGroup with 47662 atoms>
>>> u.select_atoms('around 0.0 resid 1')
Segmentation fault: 11
```

Re: actual cause of segmentation fault, I haven't looked v. closely at this but it looks like cutoff=0 triggers an infinite loop and multiple divisions-by-0 in `_NSGrid`.

https://github.com/MDAnalysis/mdanalysis/blob/e0e3462e51714f0115cf009a506a0bd352d7329a/package/MDAnalysis/lib/nsgrid.pyx#L583-L588

Reply:
Thanks, that will be useful.